include/nuttx/tls.h

```diff
@@ -205,6 +205,12 @@ struct tls_info_s
   struct pthread_cleanup_s stack[CONFIG_PTHREAD_CLEANUP_STACKSIZE];
 #endif
 
+  uint8_t tl_cpstate;                  /* Cancellation state */
```

Comment:
should we move into `#ifdef CONFIG_CANCELLATION_POINTS` too

Reply:
No, thread cancellation (asynchronous cancel) must still work even if cancellation points are disabled. So this is needed regardless